### PR TITLE
Add Python variant to jedi-base-env, update maintainers and cleanup in other bundle envs - and update spack branch from authoritative repo

### PR DESCRIPTION
**Pull request by climbfuji**
## Description
1. Add Python variant to `jedi-base-env` to allow building the JEDI software dependencies without Python packages. This means that `jedi-ewok-env` can only be built if the Python variant for `jedi-bas-env` is true (which is the default).
2. Update maintainers in all other bundle envs (from @kgerheiser to @AlexanderRichert-NOAA, from @rhoneyager to @srherbener) and perform some basic cleanup.
3. Also (unfortunately) I had to pull in updates from the spack authoritative develop branch, because the CI tests for spack no longer work (GitHub actions once again made a series of updates). I had to resolve a few conflicts, but all of them were expected (changes we made in our fork that we then sent to the authoritative repo and that got merged with a few tweaks on top of the original changes).
4. A notable change I had to make was to put `PYTHONHOME` back into the Python package, because ecflow failed to build. See https://github.com/NOAA-EMC/spack/pull/198/commits/2bd85bbe9c6a26225b70b6d50f850e2cad524054  for the commit, and https://github.com/spack/spack/issues/34028 for the bug report in the upstream spack repository.

## Testing
Tested with https://github.com/NOAA-EMC/spack-stack/pull/406

### CODE REVIEW

**Review by AlexanderRichert-NOAA — COMMENTED** [review]
Is the renaming of conftest.py related to testing problems? And is the intent to rename it later?

**Comment by climbfuji** [comment]
> Is the renaming of conftest.py related to testing problems? And is the intent to rename it later?

Good catch! I meant to move it back.

**Comment by climbfuji** [comment]
> > Is the renaming of conftest.py related to testing problems? And is the intent to rename it later?
> 
> Good catch! I meant to move it back.

Fixed. Let's see what this does to the tests.

File: var/spack/repos/jcsda-emc-bundles/packages/jedi-fv3-env/package.py [hunk 1]
```diff
@@ -12,11 +12,10 @@ class JediFv3Env(BundlePackage):
     homepage = "https://github.com/JCSDA/fv3-bundle"
     git = "https://github.com/JCSDA/fv3-bundle.git"
 
-    maintainers = ["climbfuji", "rhoneyager"]
+    maintainers = ["climbfuji", "srherbener"]
 
     version("1.0.0")
 
-    depends_on("base-env", type="run")
```
Comment (AlexanderRichert-NOAA) [inline var/spack/repos/jcsda-emc-bundles/packages/jedi-fv3-env/package.py:19]: Was this already there and wasn't necessary, or is this related to jedi-base-env changes?

Comment (climbfuji) [reply var/spack/repos/jcsda-emc-bundles/packages/jedi-fv3-env/package.py:19]: That's already included in jedi-base-env as a dependency, so not needed here.

**Review by AlexanderRichert-NOAA — APPROVED** [review]
